app/controllers/publishers_controller.rb

```diff
@@ -318,12 +344,28 @@ def show
 
   # Domain verified. See balance and submit payment info.
   def home
-    if current_publisher.promo_stats_status == :update
-      SyncPublisherPromoStatsJob.perform_later(publisher: current_publisher)
-    end
+    SyncPublisherPromoStatsJob.perform_later(publisher: current_publisher) if current_publisher.promo_stats_status == :update
+
     # ensure the wallet has been fetched, which will check if Uphold needs to be re-authorized
     # ToDo: rework this process?
-    current_publisher.wallet
+
+    wallet = current_publisher.wallet.wallet_json
+
+    # Create Uphold cards if they need to be created
+    if current_publisher.uphold_verified && current_publisher.default_currency_confirmed_at.present?
+      if should_create_default_currency_card?
+        UpholdServices::CardCreationService.new(publisher: current_publisher,
+                                                currency_code: current_publisher.default_currency).perform
+      end
+
+      if should_create_bat_card?
+        UpholdServices::CardCreationService.new(publisher: current_publisher, currency_code: "BAT").perform
+      end      
+
+      if should_update_eyeshade_default_currency?
+        PublisherDefaultCurrencySetter.new(publisher: current_publisher).perform
+      end
+    end
```

Comment:
Instead of blocking here, can we use a background job to check all publishers and create uphold cards as needed?

Reply:
This looks like it won't update the wallet in the `current_publisher` model instance. The `CardCreationService` and `PublisherDefaultCurrencySetter` will both cause changes in the wallet we may want to fetch.

Reply:
Perhaps we should introduce a new `uphold_status` value on the model that represents that these backend changes are needed and/or being processed? The status could then be polled on the controller via `PublishersController#uphold_status` and the dashboard rendering would not need to be blocked.

Reply:
@dgeb I think that's a possible solution.  If a publisher confirms their currency and we do settlement before the job runs, they won't be paid in the correct currency, though.

@lgebhardt I don't think we will need an updated wallet after this point, since the `default_currency` in the rails database is what will be displayed on the dashboard - though I could be wrong.  Is there a specific case you were thinking of?  How could we keep the wallet up-to-date?